### ISSUE TITLE
"use client" and async function breaks the page

### ISSUE THREAD

**Issue by Avratz** [vercel/next.js#50057]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Pro
    Binaries:
      Node: 18.13.0
      npm: N/A
      Yarn: N/A
      pnpm: 8.2.0
    Relevant packages:
      next: 13.4.3-canary.3
      eslint-config-next: 13.4.3
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.0.4
```

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/Avratz/testing-bug-use-client

### To Reproduce

0. Run npm run dev and go to http://localhost:3000
1. Add "use client" to page.tsx
`'use client'`
2. Make the function on the page.tsx async
`export default async function Home() {`
3. Modify some elements on page.tsx
4. The entire browser tab becomes unresponsive

### Describe the Bug

After changing an element in a page with async and "use client", the browser tab crashes. 

### Expected Behavior

Not to crash

**Comment by huozhi** [comment]
Async client components is not offically supported yet, when you remove the `"use client"` directive it becomes server component which supports async component. Checkout [the related React RFC](https://github.com/acdlite/rfcs/blob/first-class-promises/text/0000-first-class-support-for-promises.md#why-cant-client-components-be-async-functions)

**Comment by Avratz** [comment]
Hello huozhi.
Okay, I understand that. But it crashes without raising any errors or warnings. Is that behavior expected?

**Comment by theDanielJLewis** [comment]
Thanks for raising this, @Avratz! I was having the exact same problem. At least in my case, I realized I wasn't doing any `await` operations anymore, so I could remove `async` from the functions downline of my `"use client"`.

However, this makes me wonder how data should be fetched, especially from a local database, in client components. Currently, I've switched to using props. I would consider `useContext`, but that's yet another hook and so it can only be used in a client component.
